### PR TITLE
Updates to salt-junos modules

### PR DESCRIPTION
**Pull request by rsmekala**
### What does this PR do?
The PR contains new features and bug fixes for Junos modules and proxy. 

#### New features:
- Add new API get_table under junos module and state. For example,
```
# salt 'proxy01' junos.get_table FPCtable fpc_pic.yml /xxx/salt/_tables/
proxy01:
    ----------
    hostname:
        teakwood
    out:
        True
    reply:
        ----------
        1:
            ----------
            desc:
                MPCE Type 3 3D
            pic_status:
                ----------
                0:
                    ----------
                    desc:
                        10x 1GE(LAN) -E SFP
                    state:
                        Present
                1:
                    ----------
                    desc:
                        10x 1GE(LAN) -E SFP
                    state:
                        Present
```
- Added new options to `install_os` module. Options:`remote_path`, `validate`, `issu`, `nssu` 

#### Bug fixes
- Timeout issue not respected issue resolved.
- State module doesn't return success in `result` key.

### What issues does this PR fix or reference?
- https://github.com/Juniper/salt/issues/56
- https://github.com/Juniper/salt/issues/77
- https://github.com/Juniper/salt/issues/44
- https://github.com/Juniper/salt/issues/38
- https://github.com/Juniper/salt/issues/74
- https://github.com/Juniper/salt/issues/69

### Tests written?

Yes

### CODE REVIEW

**Comment by garethgreenaway** [comment]
@rsmekala Thanks for this!  Looks like there are a couple lint warnings, would you mind taking a look at those?

**Comment by rsmekala** [comment]
@garethgreenaway I have an additional package requirement `yamlordereddictloader`to be installed for CI to pass. I have already made the required changes to `requirements/tests.txt`, but it still fails with the same error.

Could you please help me through the issue.

**Comment by vnitinv** [comment]
Hi @cro Can you help us in moving the pull request to get merged. We are looking forward to have our bug fixes and new APIs added as part of this year release.

Do let us know if we have to do anything else in this process.

Thanks

**Comment by thatch45** [comment]
@rsmekala we have ordered yaml loading libs already built out in the yaml utils. I can't remember exactly how to use them but it would likely prove easier than adding a new dep, take a look at https://github.com/saltstack/salt/blob/develop/salt/utils/yamlloader.py

**Comment by rsmekala** [comment]
@thatch45 We already tried using them, but we need OrderedDict in our case and the utility `yamlloader` doesn't help in our case. I have already raised an issue for the same, https://github.com/saltstack/salt/issues/50559. I am willing to change the code once the utility is able to handle the OrderedDict. 

Could we add this dependency until the required changes are made ??

**Comment by thatch45** [comment]
Yes we can :)
I just wanted to make sure that we had looked.

@s0undt3ch what changes need to be made to get this dep working in out tests?

**Comment by s0undt3ch** [comment]
I don't see `yamlordereddictloader` in the requirements file, please add it there and I'll take care of our CI states.

**Comment by s0undt3ch** [comment]
Once https://github.com/saltstack/salt-jenkins/pull/1263 is merged, we can re-test this.

File: tests/unit/modules/test_junos.py [hunk 1]
```diff
@@ -4,11 +4,13 @@
...
 # Import python libs
 from __future__ import absolute_import, print_function, unicode_literals
+import os
 
 # Import test libs
 from tests.support.mixins import LoaderModuleMockMixin, XMLEqualityMixin
-from tests.support.mock import patch, mock_open
+from tests.support.mock import patch, mock_open, PropertyMock, call, ANY
 from tests.support.unit import skipIf, TestCase
+import six
```
Comment (s0undt3ch) [inline tests/unit/modules/test_junos.py]: Please import `salt.ext.six` instead

**Comment by s0undt3ch** [comment]
Actually, I'm not seeing that package being imported. Still necessary?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by rsmekala** [comment]
@s0undt3ch I have updated the requirements. Could you please make the changes and deploy the CI tests again

File: tests/unit/modules/test_junos.py (hunk 1 above)
Comment (rsmekala) [reply tests/unit/modules/test_junos.py]: I have updated the same in latest commit.

**Comment by s0undt3ch** [comment]
Test failure is unrelated.

**Comment by s0undt3ch** [comment]
Still, could you confirm?

**Comment by rsmekala** [comment]
@s0undt3ch The CI build seems to be passing now. Can you please resolve the conversation about `six` module.

@thatch45 Can you please approve the review for dependency so we can proceed further with the merge.

**Review by thatch45 — APPROVED** [review]

**Comment by s0undt3ch** [comment]
@thatch45 beat me to it. Thanks!

**Pull request merged**